### ISSUE TITLE
Select past EOF bug

### ISSUE THREAD

**Issue by percontation** [ridiculousfish/HexFiend#67]
The cursor may be moved beyond the end of the last byte (!) leading to nonsensical behavior and hanging.
1. In a blank document, type a few bytes. Don't completely fill the last line-wrap.
2. shift-left to select a few bytes at the end of the input
3. shift-down to modify the selection
4. keep shifting down; the cursor will fall beyond the end of the last byte. (It looks like there is no selection, but the status bar will tell you something like "7 bytes selected at offset 6 out of 6 bytes").

You can sometimes left/right arrow around in this area (and see stuff like "35 out of 15 bytes" in the status bar), and trying to type new characters hangs.

**Comment by floam** [comment]
So, the fix in 10.12 ONLY made the generation of 7F characters apparently stop happening in any kind of input field, on further inspection. You can still do this, though.

**Comment by kainjow** [comment]
When attempting to reproduce in Xcode, as soon as I hit shift-down I get this assertion:

```
Assertion failed: (HFRangeIsSubrangeOfRange(range, HFRangeMake(0, [self contentsLength]))), function -[HFController _ensureSelectionIsValid],
file /Users/me/HexFiend/framework/sources/HFController.m, line 474
```

**Comment by kainjow** [comment]
Should be fixed in df5f6e2

**Comment by floam** [comment]
Neato.
